### PR TITLE
Make s3 user agent prefix configurable

### PR DESCRIPTION
**Pull request by nezihyigitbasi**
We want to set the EC2 instance id to the user agent set on S3 requests to help us debugging issues, for that this PR makes the S3 user agent prefix configurable. The user agent suffix is still "presto". 

Related to #6079.

### CODE REVIEW

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveClientConfig.java [hunk 1]
```diff
@@ -95,6 +95,7 @@
     private DataSize s3MultipartMinPartSize = new DataSize(5, MEGABYTE);
     private boolean useParquetColumnNames;
     private boolean pinS3ClientToCurrentRegion;
+    private String s3UserAgentPrefix = "";
```
Comment (electrum) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveClientConfig.java]: This is a bit weird to have an empty string as the default for a config property. Can we make this null by default?

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveClientConfig.java [hunk 2]
```diff
@@ -772,6 +773,19 @@ public HiveClientConfig setPinS3ClientToCurrentRegion(boolean pinS3ClientToCurre
         return this;
     }
 
+    public String getS3UserAgentPrefix()
```
Comment (electrum) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveClientConfig.java]: Side note: if we were to keep the default as an empty string, this should be validated as `@NotNull`.

**Comment by electrum** [comment]
Minor comment about the default, otherwise looks good. Please assign to me when done.

**Comment by nezihyigitbasi** [comment]
If we pass `null` to the aws s3 client config then it throws when making an http request. Either we make this `null` and don't set it in the aws s3 client config in the `PrestoS3Filesystem`, or keep it as is. I thought the latter is cleaner, otherwise we will need to have `null` checks  in `PrestoS3Filesystem` and also in `HdfsConfigurationUpdater` (can't set `null` to the Hadoop `Configuration` object).

**Comment by electrum** [comment]
Merged, thanks!

**Pull request closed without merging**